learning/tour-of-beam/frontend/lib/pages/welcome/page.dart

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:app_state/app_state.dart';
+import 'package:flutter/foundation.dart';
+
+import 'screen.dart';
+import 'state.dart';
+
+class WelcomePage extends StatefulMaterialPage<void, WelcomeNotifier> {
+  static const classFactoryKey = 'Welcome';
+
+  /// Called when navigating to the page programmatically.
+  WelcomePage()
+      : super(
+          key: const ValueKey(classFactoryKey),
+          state: WelcomeNotifier(),
+          createScreen: WelcomeScreen.new,
+        );
+
+  /// Called when re-creating the page at a navigation intent.
+  // ignore: avoid_unused_constructor_parameters
+  WelcomePage.fromStateMap(Map map) : this();
```

Comment:
This constructor does not have any purpose, I think it can be removed

Reply:
These constructors are called in a chain in `PageFactory.createPage()`. It's not a factory's business to know if a certain page requires a state map to be recovered, otherwise the factory will have an extra reason to change. So it must allow each page to recover from a map. If a page does not use a map, it's its own business.